.changeset/real-drinks-melt.md

```diff
@@ -0,0 +1,30 @@
+---
+'astro': minor
+---
+
+Improved hoisted script bundling
+
+Astro's static analysis to determine which `<script>` tags to bundle together just got a little smarter!
+
+Astro create bundles that optimize script usage between pages and place them in the head of the document so that they are downloaded as early as possible. One limitation to Astro's existing approach has been that you could not dynamically use hoisted scripts. Each page received the same, all-inclusive bundle whether or not every script was needed on that page.
```

Comment:
The existing approach actually did work in most cases (delivering different bundles per page) - this change only fixes the re-export case below.

Reply:
@ottomated Can you suggest a rewording that is true? What would this paragraph have to look like so that it is accurate?

Reply:
reworded in #7733 !